app/Http/Controllers/ServidorController.php

```diff
@@ -22,10 +22,28 @@ class ServidorController extends Controller
      */
     public function index():View
     {
-        $servidores = User::where('role_id', UserRole::SERVIDOR)->paginate(10);
-        return view('servidor.index', ['servidores' => $servidores]);
+            $servidores = User::where('role_id', UserRole::SERVIDOR)->paginate(10);
+            return view('servidor.index', ['servidores' => $servidores]);
+        
     }
 
+      /**
+     * Search the specified resource from storage.
+     */
+    public function search()
+    {
+        
+        $search = request('search');
+        if($search){
+            $servidores = User::where([
+                ['name','like','%'.$search.'%']
+            ])->get();
+            return view('servidor.index', ['servidores' => $servidores,'search' => $search]);
+        }else{
+            $servidores = User::where('role_id', UserRole::SERVIDOR)->paginate(10);
```

Comment:
Aqui você está repetindo o mesmo código do método index, o que não é muito legal. Ao invés de só replicar o código, chame o `$this->index()`

Reply:
o retorno do index o redireciona para outra pagina,pois isso não seria interessante a reutilizacao de codigo dessa maneira.